auth/api/grpc/client.go

```diff
@@ -19,167 +19,71 @@ import (
 	"google.golang.org/grpc/status"
 )
 
-const svcName = "magistrala.AuthService"
+const (
+	authzSvcName  = "magistrala.AuthzService"
+	authnSvcName  = "magistrala.AuthnService"
+	policySvcName = "magistrala.PolicyService"
+)
 
-var _ magistrala.AuthServiceClient = (*grpcClient)(nil)
+var (
+	_ AuthServiceClient              = (*authGrpcClient)(nil)
+	_ magistrala.PolicyServiceClient = (*policyGrpcClient)(nil)
+)
 
-type grpcClient struct {
-	issue                endpoint.Endpoint
-	refresh              endpoint.Endpoint
-	identify             endpoint.Endpoint
-	authorize            endpoint.Endpoint
-	addPolicy            endpoint.Endpoint
-	addPolicies          endpoint.Endpoint
-	deletePolicyFilter   endpoint.Endpoint
-	deletePolicies       endpoint.Endpoint
-	listObjects          endpoint.Endpoint
-	listAllObjects       endpoint.Endpoint
-	countObjects         endpoint.Endpoint
-	listSubjects         endpoint.Endpoint
-	listAllSubjects      endpoint.Endpoint
-	countSubjects        endpoint.Endpoint
-	listPermissions      endpoint.Endpoint
-	deleteEntityPolicies endpoint.Endpoint
-	timeout              time.Duration
+//go:generate mockery --name AuthServiceClient --output=../../mocks --filename auth_client.go --quiet --note "Copyright (c) Abstract Machines"
+type AuthServiceClient interface {
+	magistrala.AuthzServiceClient
+	magistrala.AuthnServiceClient
+}
```

Comment:
Take care that you do not use AuthServiceClient where we need only AuthN or AuthZ (e.g. adapters). Always pass *minimal* dependences.